### PR TITLE
Add server abort tests

### PR DESCRIPTION
**Pull request by pakrym**
As part of https://github.com/aspnet/IISIntegration/issues/1439

I did a couple things in addition to abort tests
1. Added a pattern for writing tests that run on inproc/outofproc/outofprocv1 without deploying - we should switch to using it in tests that do not require server restarts (HelloWorld, etc.).
2. Fixed `launchSettings.json` for IIS Express
3. Added environment variable (ANCM_OUTOFPROCESS_HANDLER) for shortcircuiting out of process handler resolution and used it in launchSettings.json to point to projects build outputs and avoid copying and arranging dlls.

### CODE REVIEW

**Comment by jkotalik** [comment]
> Added environment variable (ANCM_OUTOFPROCESS_HANDLER) for shortcircuiting out of process handler resolution and used it in launchSettings.json to point to projects build outputs and avoid copying and arranging dlls.

Confused why we should add this. Is this a feature addition or something to help with testing?

**Comment by pakrym** [comment]
> Confused why we should add this. Is this a feature addition or something to help with testing?

Helps with testing, for F5 experience we don't need to copy files around and arrange them into versioned subfolders, they get used directly from project build output paths.

File: test/IISExpress.FunctionalTests/BasicAuthTests.cs [hunk 1]
```diff
@@ -15,6 +15,50 @@
 
 namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
 {
+    public abstract class ServerAbortTests: FixtureLoggedTest
```
Comment (jkotalik) [inline test/IISExpress.FunctionalTests/BasicAuthTests.cs]: nit: file is named BasicAuthTests.

**Comment by pakrym** [comment]
And by F5 experience I mean running our test sites from VS by pressing F5.

File: test/IISExpress.FunctionalTests/BasicAuthTests.cs [hunk 2]
```diff
@@ -15,6 +15,50 @@
...
+            _fixture = fixture;
+        }
+
+        [ConditionalFact]
+        public async Task ClosesConnectionOnServerAbort()
+        {
+            try
+            {
+                var response = await _fixture.Client.GetAsync("/Abort").DefaultTimeout();
+                Assert.Equal(HttpStatusCode.BadGateway, response.StatusCode);
```
Comment (jkotalik) [inline test/IISExpress.FunctionalTests/BasicAuthTests.cs]: Isn't BadGateway wrong for inproc?

File: test/IISExpress.FunctionalTests/BasicAuthTests.cs (hunk 1 above)
Comment (pakrym) [reply test/IISExpress.FunctionalTests/BasicAuthTests.cs]: Right, I'll move it out.

File: test/IISExpress.FunctionalTests/BasicAuthTests.cs (hunk 2 above)
Comment (pakrym) [reply test/IISExpress.FunctionalTests/BasicAuthTests.cs]: I'll put it under if condition.

File: test/Common.FunctionalTests/ServerAbortTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    Assert.Equal(HttpStatusCode.BadGateway, response.StatusCode);
+                    // 0x80072f78 ERROR_HTTP_INVALID_SERVER_RESPONSE The server returned an invalid or unrecognized response
+                    Assert.Contains("0x80072f78", await response.Content.ReadAsStringAsync());
+                }
+                else
+                {
+                    Assert.True(false, "Should not reach here");
+                }
+            }
+            catch (HttpRequestException)
```
Comment (jkotalik) [inline test/Common.FunctionalTests/ServerAbortTests.cs]: No way to Assert the status code for inproc? Is the status code set?

Comment (pakrym) [reply test/Common.FunctionalTests/ServerAbortTests.cs]: No, it's connection reset, that's why you get an exception.

Comment (jkotalik) [reply test/Common.FunctionalTests/ServerAbortTests.cs]: I see now. Personally I don't like this test as you don't know which variation will lead which response type. Ex Inproc and ANCMV1 will do connection reset while V2 will return 502.3. Consider splitting into two tests.

Comment (pakrym) [reply test/Common.FunctionalTests/ServerAbortTests.cs]: Both connection reset and 502 are valid for V1 and V2 outofproc.

Comment (jkotalik) [reply test/Common.FunctionalTests/ServerAbortTests.cs]: What do you get for inproc?

Comment (pakrym) [reply test/Common.FunctionalTests/ServerAbortTests.cs]: Connection reset.

Comment (jkotalik) [reply test/Common.FunctionalTests/ServerAbortTests.cs]: Can you add a comment about what are the expected results for each input?

Comment (pakrym) [reply test/Common.FunctionalTests/ServerAbortTests.cs]: Will do

**Pull request merged**